Ported/combat-bees/Assets/Scripts/Systems/CombatSystem.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Transforms;
+using Unity.Mathematics;
+
+[WithNone(typeof(Dead))]
+[BurstCompile]
+partial struct CombatJob : IJobEntity
+{
+    [ReadOnly] public ComponentLookup<LocalToWorldTransform> transformLookup;
+    public EntityCommandBuffer.ParallelWriter ECB;
+        
+    public float attackDistance;
+    
+    [BurstCompile]
+    void Execute([ChunkIndexInQuery] int chunkIndex, ref Velocity velocity, ref BeeProperties beeProperties, in LocalToWorldTransform localToWorldTransform)
+    {
+        if (beeProperties.BeeMode == BeeMode.Attack)
```

Comment:
Wouldn't it be better to filter by BeeMode.Attack outside the job via a query?

Would we just need to make BeeMode a shared component for this?